### DISCUSSION TITLE
RHF: v7.9.0 :how do I access error object with the useFormContext hook

### DISCUSSION THREAD

**Discussion by senal** [react-hook-form/react-hook-form#5765] [Q&A] [answered]
Hi,
I have 2 child components wrapped within a FormProvider.
One of them has required fields to be validated.
`    const ChildOne  = React.memo(() =>{
        const {register, formState} = useFormContext();
         return (<div>
            <input type="text" {...register('entityName', {required: "Entity name is required"})}></input>
            {formState.errors.entityName && <span>{formState.errors.entityName}</span>}
            <br/>
            <input type="text" {...register('tradingName', {required: "Trading name is required"})}></input>
        </div>)
     })`

above thrown an error 
`Error: Objects are not valid as a React child (found: object with keys {type, message, ref}). If you meant to render a collection of children, use an array instead.`

Any thoughts on this?

Appreciate it if someone can provide me with a fix for this.

Thanks

**Comment by bluebill1049** [comment, marked as answer]
You are missing formState

**Reply by senal** [reply]
Thanks @bluebill1049 ,
it seems it's working 
just did on CSB:
https://codesandbox.io/s/dreamy-monad-pmw97?file=/src/App.tsx.
Seems it's working and will with my actual code base.

Cheers

**Reply by senal** [reply]
@bluebill1049 ,
quick clarification. Is there any delay noticed getting errors with `formState`?
